components/tooltip/tooltip.directive.ts

```diff
@@ -43,37 +46,46 @@ export class TooltipDirective {
   @HostListener('focusin')
   @HostListener('mouseenter')
   public show():void {
-    if (this.visible || !this.enable) {
+    if (this.visible || !this.enable || this.delayTimeoutId) {
       return;
     }
-    this.visible = true;
-    let options = new TooltipOptions({
-      content: this.content,
-      htmlContent: this.htmlContent,
-      placement: this.placement,
-      animation: this.animation,
-      hostEl: this.viewContainerRef.element,
-      popupClass: this.popupClass,
-      context: this.tooltipContext
-    });
-
-    let binding = ReflectiveInjector.resolve([
-      {provide: TooltipOptions, useValue: options}
-    ]);
-
-    this.tooltip = this.componentsHelper
-      .appendNextToLocation(TooltipContainerComponent, this.viewContainerRef, binding);
 
-    this.triggerStateChanged();
+    this.delayTimeoutId = setTimeout(() => {
+      this.visible = true;
+      let options = new TooltipOptions({
+        content: this.content,
+        htmlContent: this.htmlContent,
+        placement: this.placement,
+        animation: this.animation,
+        hostEl: this.viewContainerRef.element,
+        popupClass: this.popupClass,
+        context: this.tooltipContext
+      });
+
+      let binding = ReflectiveInjector.resolve([
+        {provide: TooltipOptions, useValue: options}
+      ]);
+
+      this.tooltip = this.componentsHelper
+        .appendNextToLocation(TooltipContainerComponent, this.viewContainerRef, binding);
+
+      this.triggerStateChanged();
+    }, this.delay);
   }
 
   // params event, target
   @HostListener('focusout')
   @HostListener('mouseleave')
   public hide():void {
+    if (this.delayTimeoutId) {
+      clearTimeout(this.delayTimeoutId);
```

Comment:
This produces a lint error: `Assigned expression type void is not assignable to type NodeJS.Timer`